src/dbnode/encoding/iterators.go

```diff
@@ -125,6 +128,9 @@ func (i *iterators) tryAddEarliest(iter Iterator) {
 		// Reset earliest and push new iter
 		i.earliest = append(i.earliest[:0], iter)
 		i.earliestAt = dp.TimestampNanos
+		if len(firstAnnotation) > 0 {
```

Comment:
If I understand correctly, the first annotation might not be the same as the annotation of datapoint from the first `iterators::current()` call (ignoring filtering for now) depending on the `IterateEqualTimestampStrategy`. I wonder if this is not a big deal?

Reply:
Yeah, I chose to ignore this corner case, assuming that normally the annotations would be stable and consistent. I think the solution would be really complicated and not worth the effort, as there is really no good way to deal with the data if it is messed up that way.